### PR TITLE
Demonstrate API capability through demo

### PR DESCRIPTION
**Pull request by Anna-Livia**
Connected to openfisca/demonstrator#3
This PR adds a /demo page to the openfisca.fr website.
The /demo page features a simulator to calculate a couple's Disposable Income given their salary (before taxes).

This demo use the old API for now.
Next step : Integrate new API

### CODE REVIEW

File: openfisca_web_site/templates/france/fr/tree/demo.mako [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+      // Fonctions
+      function miseAjourDesSalaires(salaire, montant){
+        if (salaire == "salaire1"){
+          salaire1 = Number(montant);
+        }else if (salaire == "salaire2"){
+          salaire2 = Number(montant);
+        };
+        appelALAPI()
+      }
+      function JSONDeLaSituation(){
```
Comment (fpagnoux) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: Prefer verbs for function names, as a function usually _does_ something.
Also, prefer english to french in the code, unless we are talking about domain concepts (`ImpotSurLaFortune`, `Cotisation`)

File: openfisca_web_site/templates/france/fr/tree/demo.mako [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+                            }
+                          ]
+                        },
+                        period: "2016"
+                      }
+                    ],
+                    variables: ["revenu_disponible"],
+                    intermediate_variables: true
+          }
+        var scenarioString = JSON.stringify(scenarioObj)
```
Comment (fpagnoux) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: Unnecessary intermediate variable.

Just `return JSON.stringify(scenarioObj)`, or even 

```JS
return JSON.stringify({
    scenarios: [
    ...
})
```

File: openfisca_web_site/templates/france/fr/tree/demo.mako [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+        return scenarioString
+        }
+      function appelALAPI(){
+        var xmlhttp = new XMLHttpRequest();
+        var url = "https://api.openfisca.fr/api/1/calculate";
+        var json = JSONDeLaSituation();
+        xmlhttp.open("POST", url, true);
+
+        xmlhttp.onreadystatechange=function() {
+          //console.log(xmlhttp.responseText)
```
Comment (fpagnoux) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: Commented code should not be merged into master.

File: openfisca_web_site/templates/france/fr/tree/demo.mako [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+          if (xmlhttp.readyState==4 && xmlhttp.status==200) {
+            responseFromAPI = JSON.parse(xmlhttp.responseText);
```
Comment (fpagnoux) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: `var` missing ?

File: openfisca_web_site/templates/france/fr/tree/demo.mako [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+            transformationDeLaDonnee(responseFromAPI);
+            }
+          };
+        xmlhttp.setRequestHeader("Content-Type", "application/json");
+        xmlhttp.send(json);
+        }
+      function transformationDeLaDonnee(responseFromAPI) {
+        var results = {};
+        results["revenu_disponible"] = Math.round(responseFromAPI.value[0].menages[0].revenu_disponible["2016"]);
+        results["impots_directs"] = -Math.round((responseFromAPI.value[0].menages[0].impots_directs["2016"]));
```
Comment (fpagnoux) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: For readability, add a space before unary operators such as `-`:
```JS
- Math.round(...
```

File: openfisca_web_site/templates/france/fr/tree/demo.mako [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+        results["revenus_du_travail"] = 0;
+        for (var individu in responseFromAPI.value[0].individus){
```
Comment (fpagnoux) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: There is by construction only two `individus`. Why adding a loop ?

File: openfisca_web_site/templates/france/fr/tree/demo.mako [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+          results["revenus_du_travail"] += Math.round(responseFromAPI.value[0].individus[individu].revenus_du_travail["2016"]);
+            }
+        results["prestations_sociales"] = Math.round(responseFromAPI.value[0].familles[0].prestations_sociales["2016"]);
+        results["txDImposition"] = Math.round((results["impots_directs"])*100/(results["impots_directs"] + (results["revenu_disponible"])));
```
Comment (fpagnoux) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: Add spaces around `*` and `/` operators.

Comment (fpagnoux) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: I don't think parenthesis around ` (results["revenu_disponible"])` are necessary.

File: openfisca_web_site/templates/france/fr/tree/demo.mako [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+        results["txDisponible"] = Math.round(100 - results["txDImposition"]);
+        results["salairesBruts"] = salaire1 + salaire2;
+        results["chargesSalariales"] = results["salairesBruts"] - results["revenus_du_travail"];
+        affichageDeLInformation(results);
+        };
+      function affichageDeLInformation(results){
+        document.getElementById("txtRevenuDisponible").innerHTML=`
+          <li>dispose d'un <strong>revenu disponible</strong> de ` + (Math.round(results["revenu_disponible"])).toLocaleString("fr") + ` €/an, c'est à dire ` + (Math.round(results["revenu_disponible"]/12)).toLocaleString("fr") + ` €/mois ;</li>
```
Comment (fpagnoux) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: This is quite hard to read. I'd suggest:
- Add this in the `HTML` document further down, with `{revenu_disponible}` as a placeholder.
- In this function, replace the placeholders by the actual result.

This is basically what templating engines usually do :).

File: openfisca_web_site/templates/france/fr/tree/demo.mako (hunk 1 above)
Comment (Anna-Livia) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: 👍

File: openfisca_web_site/templates/france/fr/tree/demo.mako (hunk 2 above)
Comment (Anna-Livia) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: 👍

File: openfisca_web_site/templates/france/fr/tree/demo.mako (hunk 3 above)
Comment (Anna-Livia) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: 👍

File: openfisca_web_site/templates/france/fr/tree/demo.mako (hunk 4 above)
Comment (Anna-Livia) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: 👍

File: openfisca_web_site/templates/france/fr/tree/demo.mako (hunk 5 above)
Comment (Anna-Livia) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: 👍

File: openfisca_web_site/templates/france/fr/tree/demo.mako (hunk 6 above)
Comment (Anna-Livia) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: 👍

File: openfisca_web_site/templates/france/fr/tree/demo.mako (hunk 7 above)
Comment (Anna-Livia) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: 👍

**Comment by Anna-Livia** [comment]
I made the requested changes.
About the comments : 

"The graph is slow to load (a few seconds). Could we pre-load the results corresponding to the initial input ?"
 --> Done, but needs to be updated when legislation changes

"I don't understand this bar:" 
--> It shows the range of color in the graph. It is included in the google graph library

"I'm not fully convinced by the navigation within the graph. I click on a block, get stuck somewhere, and I don't know what to do. Do we really need this navigation ? Making it work would require some attention and time, and it seems a little more to me than what was demanded by the issue."
--> I included a "how-to" in the page but apparently, it is far from obvious.... Should we find another graph ? This one seemed the most logical to me. Suggestions ?

"When I change the input, the result evolve in a really weird way. If I increase a lot one salary, first the impots direct goes down, and then it goes up. That's odd."
--> This is due to calls to the API being made on keyup rather than on buttonclick. The solution seems to [debounce](https://css-tricks.com/debouncing-throttling-explained-examples/), but I couldn't make it work. Quick fix : Bring back the button / Long fix : I need help integrating Debounce

File: openfisca_web_site/templates/france/fr/tree/demo.mako [hunk 9]
```diff
@@ -0,0 +1,245 @@
...
+      function callOpenFiscaAPI(){
+        var xmlhttp = new XMLHttpRequest();
+        var url = "http://localhost:5000/calculate";
+        var json = formatJSONOfSituation();
+        xmlhttp.open("POST", url, true);
+
+        xmlhttp.onreadystatechange=function() {
+          if (xmlhttp.readyState==4 && xmlhttp.status==200) {
+            var responseFromAPI = JSON.parse(xmlhttp.responseText);
+            transformData(responseFromAPI);
```
Comment (fpagnoux) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: This architecture is not great : every function must know what is the next function to call... and it's not natural for a function named `callOpenFiscaAPI ` to end up calling a function that process results, and another that display them.

The JS way of handling this is callbacks:

```JS
function callOpenFiscaAPI(callback) {
    ...
    var responseFromAPI = JSON.parse(xmlhttp.responseText);
    callback(responseFromAPI)
}
```

and in `updateSalaries `:
```JS
callOpenFiscaAPI(function(response) {
    results = transformData(response)
    displayData(results)
});
```

File: openfisca_web_site/templates/france/fr/tree/demo.mako [hunk 10]
```diff
@@ -0,0 +1,245 @@
...
+              },
+              "revenu_disponible": {
+                "2017": null
+              }
+            }
+          }
+        });
+      }
+
+      function callOnLoad(){
```
Comment (fpagnoux) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: Function names should help us understand what they do at a glance. 

When I read `window.addEventListener("load", callOnLoad, false);`, I understand : when the window is loaded... call the function to call when the window is load. Which doesn't help 😆 

What about calling this function `initPage` ? or `preDisplayPage` ? This way the previous code becomes : 
`window.addEventListener("load", initPage, false);`. The intention is much clearer : when the window is loaded, initialise the page.

File: openfisca_web_site/templates/france/fr/tree/demo.mako [hunk 11]
```diff
@@ -0,0 +1,245 @@
...
+from openfisca_web_site import conf
+from openfisca_web_site import urls
+%>
+
+<%inherit file="/page.mako"/>
+
+<%def name="scripts()" filter="trim">
+    <script src="${urls.get_static_url(ctx, u'/bower/lodash/dist/lodash.min.js')}"></script>
+    <script src="https://code.jquery.com/jquery-3.2.1.min.js"
+    integrity="sha256-hwg4gsxgFZhOsEEamdOYGBf13FyQuiTwlAQgxVSNgt4="
```
Comment (fpagnoux) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: Not enough indented.

File: openfisca_web_site/templates/france/fr/tree/demo.mako [hunk 12]
```diff
@@ -0,0 +1,245 @@
...
+
+<%def name="page_content()" filter="trim">
+          <p class="lead">Le revenu disponible représente l'ensembles des revenus issus du travail, des pensions, des revenus du capital et des prestations sociales, auxquelles on soustrait les impots directs du ménage.</p>
+
+      <div class="row">
+        <!-- entrée des données -->
+        <div class="col-md-4">
+          <form action="javascript:void(0);">
+            <p><b>Salaire annuel brut du premier conjoint :</b></p>
+            <p>
```
Comment (fpagnoux) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: You're opening a paragraph without closing it, that's not valid HTML ! 
If you want to add space after a `p`, add a `margin` in CSS.

File: openfisca_web_site/templates/france/fr/tree/demo.mako [hunk 13]
```diff
@@ -0,0 +1,245 @@
...
+            <p>
+            <input type="text" onkeyup=updateSalaries("salary1",this.value) size="20" value="25000"> €/an
+            </p>
+            <p><b>Salaire annuel brut du second conjoint :</b></p>
+            <p>
+            <input type="text" onkeyup=updateSalaries("salary2",this.value) size="20" value="15000"> €/an
+            </p>
+            <p>
```
Comment (fpagnoux) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: Empty paragraph. If the goal is to add space, use `margin`. In general, styling must be done trough CSS, and not by adding non-semantic stuff to the HTML. It's not clean, and makes the website less accessible.

File: openfisca_web_site/templates/france/fr/tree/demo.mako [hunk 14]
```diff
@@ -0,0 +1,245 @@
...
+          <p class="lead">Le revenu disponible représente l'ensembles des revenus issus du travail, des pensions, des revenus du capital et des prestations sociales, auxquelles on soustrait les impots directs du ménage.</p>
+
+      <div class="row">
+        <!-- entrée des données -->
+        <div class="col-md-4">
+          <form action="javascript:void(0);">
+            <p><b>Salaire annuel brut du premier conjoint :</b></p>
+            <p>
+            <p>
+            <input type="text" onkeyup=updateSalaries("salary1",this.value) size="20" value="25000"> €/an
```
Comment (fpagnoux) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: Identation

File: openfisca_web_site/templates/france/fr/tree/demo.mako [hunk 15]
```diff
@@ -0,0 +1,245 @@
...
+            </p>
+            <p><b>Salaire annuel brut du second conjoint :</b></p>
+            <p>
+            <input type="text" onkeyup=updateSalaries("salary2",this.value) size="20" value="15000"> €/an
```
Comment (fpagnoux) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: Identation

File: openfisca_web_site/templates/france/fr/tree/demo.mako [hunk 16]
```diff
@@ -0,0 +1,245 @@
...
+          <span id="situationDescription"></span>
+        </div>
+      </div>
+      <hr>
+      <div class="row">
+        <!-- entrée des données -->
+        <div class="col-md-4">
+        <h3>Visualisation</h3>
+          <ul>
+          <li>Cliquez sur "Revenu disponible" pour en voir le détail.</li>
```
Comment (fpagnoux) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: Identation

File: openfisca_web_site/templates/france/fr/tree/demo.mako [hunk 17]
```diff
@@ -0,0 +1,245 @@
...
+          </ul>
+          </div>
+       <!-- Visualisation -->
+        <div class="col-md-8">
+          <!-- partie sur les revenus disponibles -->
+          <span id="graph"></span>
+          </div>
+        </div>
+        <div class="row">
+        <div class="col-md-12">
```
Comment (fpagnoux) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: Identation

File: openfisca_web_site/templates/france/fr/tree/demo.mako [hunk 18]
```diff
@@ -0,0 +1,245 @@
...
+        <hr>
```
Comment (fpagnoux) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: Identation

Comment (Anna-Livia) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: 👍

File: openfisca_web_site/templates/france/fr/tree/demo.mako (hunk 17 above)
Comment (Anna-Livia) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: 👍

File: openfisca_web_site/templates/france/fr/tree/demo.mako (hunk 16 above)
Comment (Anna-Livia) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: 👍

File: openfisca_web_site/templates/france/fr/tree/demo.mako (hunk 13 above)
Comment (Anna-Livia) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: 👍

File: openfisca_web_site/templates/france/fr/tree/demo.mako (hunk 15 above)
Comment (Anna-Livia) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: 👍

File: openfisca_web_site/templates/france/fr/tree/demo.mako (hunk 12 above)
Comment (Anna-Livia) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: 👍

File: openfisca_web_site/templates/france/fr/tree/demo.mako (hunk 11 above)
Comment (Anna-Livia) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: Too much indentation, isn't it ?

File: openfisca_web_site/templates/france/fr/tree/demo.mako (hunk 10 above)
Comment (Anna-Livia) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: :+

File: openfisca_web_site/templates/france/fr/tree/demo.mako [hunk 19]
```diff
@@ -0,0 +1,245 @@
...
+        </div>
+        <div class="col-md-8">
+          <span id="graph"></span>
+        </div>
+      </div>
+      <!-- Version d'OpenFisca -->
+      <div class="row">
+        <div class="col-md-12">
+          <hr>
+          <p>OpenFisca-France@18.6</p>
```
Comment (fpagnoux) [inline openfisca_web_site/templates/france/fr/tree/demo.mako]: That could be a big and mean lie. if we're not able to get the version number right now, just don't mention the version number ;)

**Pull request merged**